src/fenicsxconcrete/problem.py

```diff
@@ -0,0 +1,73 @@
+from pydantic.dataclasses import dataclass
+
+from pydantic import RootModel
+
+from bcs import (
+    BodyForceDefinition,
+    DirichletBCDefinition,
+    InitialConditionDefinition,
+    NeumannBCDefinition,
+)
+from material import LinearElasticMaterial
+from mesh import MeshGenerator
+
+from sensor import PlotDefinition, PointSensorDefinition, GlobalSensorDefinition
+from experiment import Experiment
+
+@dataclass(config=dict(arbitrary_types_allowed=True))
+class QuadratureRuleDefinition:
+    name: str
+    order: int
+    variable: str
+
+@dataclass(config=dict(arbitrary_types_allowed=True))
+class FiniteElementDefinition:
+    name: str
+    geometry_order: int
+    function_order: int
+    variable: str
+
+@dataclass(config=dict(arbitrary_types_allowed=True))
+class FEMProblemDefinition:
+    experiment: Experiment
+    sensors: list[PointSensorDefinition | GlobalSensorDefinition]
+    plots: list[PlotDefinition]
+    element_type: list[FiniteElementDefinition]
+    quadrature_rule: list[QuadratureRuleDefinition]
+
```

Comment:
Here, the different Experiment and Definition objects would only be used for initializing the problem and then can be forgotten, right? That would work with the model updating schemes, as then I can access directly the variables from FEMProblem (or the derive class) and update their values.

Reply:
I would say, that most attributes could be forgotten after initialization. However, I haven't really considered the implementation part of the problems yet.